### ISSUE TITLE
Cannot decode type 'bytes'

### ISSUE THREAD

**Issue by DavidNeuroth** [lidatong/dataclasses-json#365]
In the following script decoding from json fails:
```python
from dataclasses import dataclass
from dataclasses_json import dataclass_json

@dataclass_json
@dataclass
class Test:
    x: bytes = b""

myobj = Test()
s = myobj.to_json()

# this line fails
result = Test.from_json(s)
```

The following exception is thrown:
```
AttributeError: type object 'bytes' has no attribute '__args__'
```

Observed with Python 3.10.4 and dataclasses_json 0.5.7.

**Comment by Evgeniy-Sinyak** [comment]
I've encountered a similar issue. I am using version 0.5.7 of the `dataclasses-json` library, and I have noticed that the bytes type is not included in the list of supported types listed [here](https://lidatong.github.io/dataclasses-json/). It would be more convenient if the library failed during serialization rather than deserialization, as it would be easier to identify and handle the issue at an earlier stage. It shouldn't silently convert bytes into array since it can't parse it back.

**Comment by george-zubrienko** [comment]
I don't think bytes should be allowed without custom encoder/decoder, wdyt @matt035343 ?

**Comment by lidatong** [comment]
> it would be more convenient if the library failed during serialization rather than deserialization

Yeah this is an interesting idea. I see the benefit of fail-fast, but use-cases differ and the best would be each user can parameterize their own strategy.

A 3rd-party contributor wrote this PR to do that #159 which was really useful. It's also documented in README. Does this `bytes` issue play nicely with that? I haven't tried myself.

**Comment by matt035343** [comment]
I don't think it is clear from the issue description if what the expected behaviour is. 
However,
1. I agree with @Evgeniy-Sinyak that if it should fail, then it should do it already in serialization
2. I agree with @george-zubrienko that bytes should not be supported by default as JSON does not support such serialization natively - custom encodes/decoders exist for this
3. I agree with @lidatong that the way forward is to see if #159 can be extended handle bytes as well

**Comment by matt035343** [comment]
@george-zubrienko It seems like this has been fixed in v0.5.8 probably in PR #346.

Closing.
